kart/tabular/ogr_import_source.py

```diff
@@ -132,6 +133,13 @@ def open(cls, source, table=None, primary_key=None):
         try:
             ds = cls._ogr_open(ogr_source, **open_kwargs)
         except RuntimeError as e:
+            # Check whether the source is a shapefile, and if so, whether it's missing an .shx file
+            if source.lower().endswith(".shp"):
+                base_filename = os.path.splitext(os.path.basename(source))[0]
+                raise NotFound(
```

Comment:
you haven't actually checked for the missing .shx file here. You'll need to either parse the RuntimeError's message, or look on disk for the file to see if it exists